plugin/vscode/quick-lint-js.js

```diff
@@ -160,9 +162,14 @@ class Parser {
     }
   }
 
-  lint() {
+  lint(performanceWriter) {
+    let lint_start;
+    let lint_end;
+    lint_start = process.hrtime.bigint()
     let diagnosticsPointer = this._process._lint(this._parser);
-
+    lint_end = process.hrtime.bigint()
+    performanceWriter.fileNameOptimizer()
```

Comment:
Shouldn't we only need to do this when we open the document, not every time we lint?

Reply:
When I asked you initially about logging, you mentioned that we wanted to log every time we lint (i.e. a keystroke is pressed). Could you expand more on what you mean by "when we open the document"?

Reply:
> Could you expand more on what you mean by "when we open the document"?

Basically, the first lint for that file; the first time the quick-lint-js extension becomes aware of a file.